dask/dataframe/core.py

```diff
@@ -4789,8 +4789,10 @@ def maybe_shift_divisions(df, periods, freq):
 
 
 @wraps(pd.to_datetime)
-def to_datetime(arg, **kwargs):
-    meta = pd.Series([pd.Timestamp('2000')])
+def to_datetime(arg, meta=None, **kwargs):
+    if meta is None:
+        meta = pd.Series([pd.Timestamp('2000')])
+        meta.index.name = arg.index.name
```

Comment:
This is buggy on master, but you can pass a dd.Index here,

```python
In [13]: s = pd.Series(1, index=idx)

In [14]: idx = pd.Index(pd.date_range('2000', periods=12).astype(str))

In [15]: s = pd.Series(1, index=idx)

In [16]: ds = dd.from_pandas(s, 2)

In [17]: dd.to_datetime(ds.index).compute()
Out[17]:
DatetimeIndex(['2000-01-01', '2000-01-02', '2000-01-03', '2000-01-04',
               '2000-01-05', '2000-01-06', '2000-01-07', '2000-01-08',
               '2000-01-09', '2000-01-10', '2000-01-11', '2000-01-12'],
              dtype='datetime64[ns]', freq=None)

In [18]: dd.to_datetime(ds.index)
Out[18]:
Dask Series Structure:
npartitions=2
2000-01-01    datetime64[ns]
2000-01-07               ...
2000-01-12               ...
dtype: datetime64[ns]
Dask Name: to_datetime, 6 tasks
```

with these changes, it raises on the `arg.index.name`. I'd recommend only setting the name here if the `arg` has a name attribute.

You're welcome to add tests for `meta=pd.Index` (everything may work now), but of course no obligation.

Reply:
ok tried to account for that and add tests. Had to hack a bit as `check_divisions` within `assert_eq` seemed to fail because it was comparing to a `NaT` division. And `sort=False` was important because `assert_eq` for indices uses pandas index comparison tools